components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,528 @@
+        Assert.assertEquals(AuthType.AuthenticationType.BASIC, result.getEndpoint().getAuthentication().getType());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getProperties().get(0).getValue(),
+                result.getEndpoint().getAuthentication().getProperties().get(0).getValue());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getProperties().get(1).getValue(),
+                result.getEndpoint().getAuthentication().getProperties().get(1).getValue());
+    }
+
+    @Test(priority = 11)
+    public void testUpdateActionWithoutEndpointUri() throws ActionMgtException {
+
+        Action updatingAction = buildMockAction(
+                "Pre Issue Access Token",
+                "To configure pre issue access token",
+                null,
+                 AuthType.AuthenticationType.BASIC,
+                 buildMockBasicAuthProperties("updatingadmin", "updatingadmin"));
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 12)
+    public void testUpdateActionWithAuthType() throws ActionMgtException {
+
+        Action updatingAction = buildMockAction(
+                "Pre Issue Access Token",
+                "To configure pre issue access token",
+                "https://sample.com",
+                AuthType.AuthenticationType.BEARER,
+                buildMockBearerAuthProperties("57c7df90-cacc-4f56-9b0a-f14bfbff3076"));
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType());
+        action = result;
+    }
+
+    @Test(priority = 13)
+    public void testUpdateActionWithUri() throws ActionMgtException {
```

Comment:
This is again and error case. Let's add a todo

Reply:
Updated in ecbc2c308facba3438fc42a2ab37e49cb46d1591